tidy3d/components/data/sim_data.py

```diff
@@ -792,6 +792,28 @@ def plot_scalar_array(
 
         return ax
 
+    def to_mat(self, fname):
```

Comment:
type annotation and could you put the variable stuff in the docstring (eg `SimulationData`, `.mat`) in double backslash quotes? (side note: I forget this all the time, it's why we have a 2nd or 3rd pair of eyes on these PRs).

Reply:
Also, after reading through, I wonder if we should rename this `to_mat_file()` just to be a bit more explicit. Sorry to switch it up last minute, but what do you think?

Reply:
We could also consider making this `savemat`, just like `scipy`.

Reply:
speaking of this, it could be useful to just accept also some `**kwargs` in the function, which can then be passed to `scipy.io.savemat` if the user wants. For example

```py
def savemat(file_name, **kwargs):
    
     if "m_dict" in kwargs:
        raise ValueError("m_dict" is automatically determined by the object, can't pass to `savemat`.)

     m_dict = ...

     return scipy.io.savemat(file_name, m_dict, **kwargs)

```